utilities/transactions/transaction_base.h

```diff
@@ -108,6 +114,12 @@ class TransactionBaseImpl : public Transaction {
                              keys, values);
   }
 
+  void MultiGetSingleCFForUpdate(const ReadOptions& options,
+                                 ColumnFamilyHandle* column_family,
+                                 const size_t num_keys, const Slice* keys,
+                                 PinnableSlice* values, Status* statuses,
+                                 bool sorted_input = false) override;
```

Comment:
Does MyRocks plan to use this function?

Reply:
Yes, they use GetForUpdate() today and wanted something similar for MultiGet. The Get time in GetForUpdate() is a bottleneck in fastpathdb. Do you see the lack of multi-column family support here as an issue?

Reply:
No problem. I just feel that it is complicated to design the function behavior while we need to lock multiple keys. Right now, if I understand correctly, if one lock fails to be acquired, we don't release the ones being locked, so we have partial keys locked when the function returns. It's not necessarily wrong, but we we need to think more about whether it is the API we want to be. If we don't need to implement MultiGetSingleCFForUpdate(), then we can avoid the trouble.

Reply:
@siying @anand1976 I don't see how we can easily group the MultiGet/MultiGetForUpdate in the regular MyRocks code path without non-trivial MySQL side changes. In the new query engine we have good opportunity to use MultiGet for queries such as IN ranges which is what I'm benchmarking right now. But there are no immediate plans to support the ForUpdate version there.

Reply:
@siying @yizhang82 If there are no immediate plans to use MultiGetForUpdate(), let me defer it to a separate PR so we can come up with the right API.